src/services/identity/postgres.ts

```diff
@@ -232,36 +240,69 @@ export class PostgresIdentityService extends DefaultIdentityService {
 
 	async createStatusList2021(did: string, resourceOptions: ResourcePayload, statusOptions: CreateStatusListOptions, agentId: string): Promise<CreateStatusList2021Result> {
 		const agent = await this.createAgent(agentId)
+        if (!await CustomerService.instance.find(agentId, { did })) {
+            throw new Error(`${did} not found in wallet`)
+        }
 		return await Veramo.instance.createStatusList2021(agent, did, resourceOptions, statusOptions)
 	}
 
 	async updateStatusList2021(did: string, statusOptions: UpdateStatusListOptions, publish: boolean, agentId: string): Promise<BulkRevocationResult | BulkSuspensionResult | BulkUnsuspensionResult> {
 		const agent = await this.createAgent(agentId)
+        if (!await CustomerService.instance.find(agentId, { did })) {
+            throw new Error(`${did} not found in wallet`)
+        }
 		return await Veramo.instance.updateStatusList2021(agent, did, statusOptions, publish)
 	}
 
 	async checkStatusList2021(did: string, statusOptions: CheckStatusListOptions, agentId: string): Promise<StatusCheckResult> {
 		const agent = await this.createAgent(agentId)
+        if (!await CustomerService.instance.find(agentId, { did })) {
```

Comment:
I think for checking the statusList we don't need to enable such guarding cause it's public info. Am I right?